### PR TITLE
[inference] add llama mlp for smoothquant

### PR DESCRIPTION
**Pull request by Xu-Kai**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?
- Add llama mlp for smoothquant
- Refactor llama code

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/inference/quant/smoothquant/models/__init__.py [hunk 1]
```diff
@@ -9,4 +9,4 @@
     )
 
 if HAS_TORCH_INT:
-    from .smoothquant_layer import LLamaSmoothquantAttention
```
Comment (yuanheng-zhao) [inline colossalai/inference/quant/smoothquant/models/__init__.py:12]: Are you going to remove the corresponding lines in `smooth quant_layer`?

File: tests/test_smoothquant/test_llama_mlp.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    smoothquant_cuda = SmoothquantBuilder().load()
+    HAS_SMOOTHQUANT_CUDA = True
+except ImportError:
+    warnings.warn("CUDA smoothquant linear is not installed")
+    HAS_SMOOTHQUANT_CUDA = False
+
+from colossalai.inference.quant.smoothquant.models import LlamaSmoothquantMLP
+
+try:
+    from colossalai.inference.quant.smoothquant.models import LlamaSmoothquantMLP
```
Comment (yuanheng-zhao) [inline tests/test_smoothquant/test_llama_mlp.py:19]: remove duplicates

File: colossalai/inference/quant/smoothquant/models/llama.py [hunk 3]
```diff
@@ -183,3 +189,44 @@ def forward(
...
+    ):
+        int8_module = LlamaSmoothquantMLP(
+            mlp_module.intermediate_size,
+            mlp_module.hidden_size,
+        )
+
+        int8_module.gate_proj = W8A8BFP32O32LinearSiLU.from_float(mlp_module.gate_proj, gate_proj_input_scale)
+        int8_module.up_proj = W8A8BFP32OFP32Linear.from_float(mlp_module.up_proj, up_proj_input_scale)
+        int8_module.down_proj = W8A8BFP32OFP32Linear.from_float(mlp_module.down_proj, down_proj_input_scale)
+        self.down_proj_input_scale = down_proj_input_scale
```
Comment (yuanheng-zhao) [inline colossalai/inference/quant/smoothquant/models/llama.py:217]: Wonder where will this variable `self.down_proj_input_scale` be used

**Review by yuanheng-zhao — APPROVED** [review]
Left some comments

File: tests/test_smoothquant/test_llama_mlp.py (hunk 2 above)
Comment (Xu-Kai) [reply tests/test_smoothquant/test_llama_mlp.py]: Removed.

File: colossalai/inference/quant/smoothquant/models/llama.py (hunk 3 above)
Comment (Xu-Kai) [reply colossalai/inference/quant/smoothquant/models/llama.py:217]: Deleted the inter_out_scale and use self.down_proj_input_scale.

**Pull request merged**